src/Soil/soil_hydrology_parameterizations.jl

```diff
@@ -41,84 +59,111 @@ function volumetric_liquid_fraction(ϑ_l::FT, ν_eff::FT, θ_r::FT) where {FT}
     return θ_l
 end
 
+"""
+    effective_saturation(porosity::FT, ϑ_l::FT, θr::FT) where {FT}
+
+A point-wise function computing the effective saturation.
+"""
+function effective_saturation(porosity::FT, ϑ_l::FT, θr::FT) where {FT}
+    ϑ_l_safe = max(ϑ_l, θr + eps(FT))
+    S_l = (ϑ_l_safe - θr) / (porosity - θr)
+    return S_l
+end
+
 
 """
-     matric_potential(α::FT, n::FT, m::FT, S::FT) where {FT}
+     matric_potential(cm::vanGenuchten{FT}, S::FT) where {FT}
 
 A point-wise function returning the matric potential, using the
 van Genuchten formulation.
 """
-function matric_potential(α::FT, n::FT, m::FT, S::FT) where {FT}
+function matric_potential(cm::vanGenuchten{FT}, S::FT) where {FT}
+    (; α, m, n) = cm
     ψ = -((S^(-FT(1) / m) - FT(1)) * α^(-n))^(FT(1) / n)
     return ψ
 end
 
 """
-     inverse_matric_potential(α::FT, n::FT, m::FT, ψ::FT) where {FT}
+     inverse_matric_potential(cm::vanGenuchten{FT}, ψ::FT) where {FT}
 
 A point-wise function returning the effective saturation, given
 the matric potential, using the
 van Genuchten formulation.
 """
-function inverse_matric_potential(α::FT, n::FT, m::FT, ψ::FT) where {FT}
+function inverse_matric_potential(cm::vanGenuchten{FT}, ψ::FT) where {FT}
     ψ > 0 && error("Matric potential is positive")
+    (; α, m, n) = cm
     S = (FT(1) + (α * abs(ψ))^n)^(-m)
     return S
 end
 
-"""
-    effective_saturation(porosity::FT, ϑ_l::FT, θr::FT) where {FT}
 
-A point-wise function computing the effective saturation.
-"""
-function effective_saturation(porosity::FT, ϑ_l::FT, θr::FT) where {FT}
-    ϑ_l_safe = max(ϑ_l, θr + eps(FT))
-    S_l = (ϑ_l_safe - θr) / (porosity - θr)
-    return S_l
-end
 
 """
     pressure_head(
-        α::FT,
-        n::FT,
-        m::FT,
+        cm::vanGenuchten,
         θ_r::FT,
         ϑ_l::FT,
         ν_eff::FT,
         S_s::FT,
     ) where {FT}
 
 A point-wise function returning the pressure head in
-variably saturated soil, using the van Genuchten formulation
-for matric potential if the soil is not saturated, and
+variably saturated soil, using the van Genuchten matric potential 
+if the soil is not saturated, and
 an approximation of the positive pressure in the soil if the
 soil is saturated.
 """
 function pressure_head(
-    α::FT,
-    n::FT,
-    m::FT,
+    cm::vanGenuchten,
```

Comment:
Should this dispatch on `::vanGenuchten{FT}`? I think it's better to specify the subtype of the struct as well